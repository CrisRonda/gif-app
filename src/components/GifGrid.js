import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GiftGridItem from "./GiftGridItem";

import PropTypes from "prop-types";

const GifGrid = ({ category, limit }) => {
  const { data: images, loading } = useFetchGifs({ category, limit });

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">
        {category.toUpperCase()}
      </h3>
      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}
      <main className="card-grid">
        {images.map((img) => (
          <GiftGridItem key={img.id} {...img} />
        ))}
      </main>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
  limit: PropTypes.any.isRequired,
};
export default GifGrid;
