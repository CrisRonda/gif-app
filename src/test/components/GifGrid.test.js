import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Test on GifGrid", () => {
  const category = "Goku";
  const limit = 3;
  test("should render compontent", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    let wrapper = shallow(<GifGrid category={category} limit={limit} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render n items when end fetch data", () => {
    const data = [
      {
        id: "123",
        title: "cualquier cosa",
        url:
          "https://s3.amazonaws.com/uifaces/faces/twitter/aaronalfred/128.jpg",
      },
      {
        id: "123",
        title: "cualquier cosa",
        url:
          "https://s3.amazonaws.com/uifaces/faces/twitter/aaronalfred/128.jpg",
      },
      {
        id: "123",
        title: "cualquier cosa",
        url:
          "https://s3.amazonaws.com/uifaces/faces/twitter/aaronalfred/128.jpg",
      },
      {
        id: "123",
        title: "cualquier cosa",
        url:
          "https://s3.amazonaws.com/uifaces/faces/twitter/aaronalfred/128.jpg",
      },
    ];
    useFetchGifs.mockReturnValue({
      data,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} limit={limit} />);
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GiftGridItem").length).toBe(data.length);
  });
});
