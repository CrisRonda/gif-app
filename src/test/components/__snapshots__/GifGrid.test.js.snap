// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifGrid should render compontent 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    GOKU
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <main
    className="card-grid"
  />
</Fragment>
`;
